internal/backend/db/dbgorm/connections.go

```diff
@@ -21,12 +22,30 @@ func (gdb *gormdb) createConnection(ctx context.Context, conn *scheme.Connection
 	return gdb.createEntityWithOwnership(ctx, createFunc, conn, conn.ProjectID)
 }
 
+func (gdb *gormdb) deleteConnectionsAndVars(where string, args ...interface{}) error {
```

Comment:
but what is args ?
it's a delete statement without a proper signature
feels a bit scary

Reply:
it's an internal generic method. And we have this in gormdb in generic delete forexample which accepts exactly the same params, e.g. where + args.

Reply:
we have it doesn't mean it is a great thing

WDYT about something like this? still generic, but not total open to anything

```go
type queryAugmenter func(q *gorm.DB) *gorm.DB

func (gdb *gormdb) deleteProjectConnections() {
	gdb.deleteConnectionsAndVars(func(q *gorm.DB) *gorm.DB {
		return q.Where("project_id = ?", "my_porject")
	})
}

func (gdb *gormdb) deleteConnectionsAndVars(where queryAugmenter) {
	q := gdb.db
	q = where(q)
	q.Delete(&scheme.Connection{})
}

```

Reply:
I'm sorry Efi, but I'm probably missing the point. Could u explain me why this is better? I'm passing the same "project_id = ?" and projectID as a parameter. And in the second case "connection_id =?" and connection_id. I could add those scope functions, and use it with but I very puzzled, since I see no benefits neither in usability nor in security. 
- In our version you are passing the func with hardcoded `Where` parameters. 
- In my case I'm passing hardcoded `parameters` and doing `Where` in place. 

Could you plz once again explain why you think it's a problem to do exactly what we are doing in other places of our code?

Reply:
I think the point Efi is making is that there's a difference between the lack of safety in Gorm's `Delete` (which is by design for maximum flexibility at the lowest level), and the lack of safety in this higher-level function.

In our case we don't want to offer maximum flexibility, we want to ensure the caller knows exactly what they're doing, and can't do what we didn't plan to allow them. Type safety is a way to achieve that.

If you don't want separate functions like Efi suggested, you can also use a single function with a variadic option pattern - `opts` is type-safe, but allows a variable number of customizations. For example:

https://github.com/autokitteh/autokitteh/blob/main/sdk/sdkintegrations/integration.go#L59

...But personally I think it's more complicated and less useful for just 2 options.

Reply:
I changed type from interface{} to sdktypes.UUID